Migrate to Bazel 8

dorezyuk opened the pull request:
Goes together with 
https://github.com/EVerest/everest-utils/pull/187
https://github.com/EVerest/everest-core/pull/1180

Commit:
Bazel 8

Signed-off-by: Dima Dorezyuk <ddo@qwello.eu>

Commit:
revert workflow change

Signed-off-by: Dima Dorezyuk <ddo@qwello.eu>

hikinggrass commented on BUILD.bazel:
```diff
@@ -94,12 +94,16 @@ cc_binary(
 )
+
+exports_files([
+   "dependencies.yaml",
```
is this still used when the dependencies.yaml parsing is dropped for now?

hikinggrass replied:
(ok, still used in everest-core)

hikinggrass commented on MODULE.bazel:
```diff
@@ -0,0 +1,119 @@
+module(
+    name = "everest-framework",
+    version = "0.0.0",
```
we should add this to the [check_version_consistency.py](https://github.com/EVerest/everest-framework/blob/main/.ci/build-kit/scripts/check_version_consistency.py) to keep it in sync with what we have in CMake.
I can do that quickly for this PR

hikinggrass committed:
Check MODULE.bazel version for consistency as well

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Set MODULE.bazel everest-framework module version to 0.20.4

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
WORKSPACE.bazel has been removed, use MODULE.bazel instead in bazel cache key

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Add extra rustc flags for using gold as linker to .bazelrc

Not setting this would make the bazel build fail on some systems (eg. opensuse)

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

dorezyuk committed:
Update .bazelrc

Signed-off-by: Dima Dorezyuk <dmitrij.dorezyuk@hotmail.de>
